packages/flutter/lib/src/widgets/_window.dart

```diff
@@ -1102,6 +1264,68 @@ class TooltipWindow extends StatelessWidget {
   }
 }
 
+/// The [PopupWindow] widget provides a way to render a popup window in the
+/// widget tree.
+///
+/// The provided [controller] creates the native window that backs
+/// the widget. The [child] widget is rendered into this newly created window.
+///
+/// When a [PopupWindow] widget is removed from the tree, the window that was created
+/// by the [controller] remains valid until the caller destroys it by calling
+/// [PopupWindowController.destroy].
+///
+/// Widgets in the same tree as the [child] widget will have access to the
+/// [PopupWindowController] via the [WindowScope] widget.
+///
+/// {@tool dartpad}
```

Comment:
This will run the example live in Dartpad on the docs website. I'm guessing this is not useful because it won't work on the web right? So I would change this to `@tool snippet`.